File: app/src/main/java/com/njust/helper/course/fragment/CourseDayAdapter.java
package com.njust.helper.course.fragment;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.njust.helper.databinding.ItemCourseDayBinding;
import com.njust.helper.model.Course;
import com.njust.helper.tools.DataBindingHolder;

import java.util.List;

public class CourseDayAdapter extends RecyclerView.Adapter<DataBindingHolder<ItemCourseDayBinding>> {
    @SuppressWarnings("unchecked")
    private List<Course>[] mData = new List[0];
    private int mWeek = 0;
    private int mDay = 0;
    private CourseDayFragment fragment;

    CourseDayAdapter(CourseDayFragment fragment) {
        this.fragment = fragment;
    }

    public void setData(@NonNull List<Course>[] data, int week, int day) {
        mData = data;
        mWeek = week;
        mDay = day;
        notifyDataSetChanged();
    }

    @Override
    public DataBindingHolder<ItemCourseDayBinding> onCreateViewHolder(ViewGroup parent, int viewType) {
        ItemCourseDayBinding binding = ItemCourseDayBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new DataBindingHolder<>(binding);
    }

    @Override
    public void onBindViewHolder(DataBindingHolder<ItemCourseDayBinding> holder, int position) {
        final List<Course> list = mData[position];
        final ItemCourseDayBinding binding = holder.getDataBinding();
        binding.setEmpty(list.size() == 0);
        binding.setMultiple(list.size() > 1);
        binding.setPosition(position);
        String weekString = " " + mWeek + " ";
        if (list.size() > 0) {
            Course course = null;
            for (Course t : list) {
                course = t;
                if (course.getWeek2().contains(weekString))
                    break;
            }
            assert course != null;
            binding.setValid(course.getWeek2().contains(weekString));
            binding.setCourse(course);
            binding.getRoot().setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    fragment.getListener().showCourseList(list, mDay, binding.getPosition());
                }
            });
        } else {
            binding.getRoot().setOnClickListener(null);
        }
    }

    @Override
    public int getItemCount() {
        return mData.length;
    }
}

File: app/src/main/java/com/njust/helper/course/day/CourseDayAdapter.kt
package com.njust.helper.course.day

import android.view.ViewGroup
import androidx.compose.ui.platform.ComposeView
import androidx.recyclerview.widget.RecyclerView
import com.njust.helper.model.Course

class CourseDayAdapter internal constructor(private val fragment: CourseDayFragment)
  : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
  private var mData: Array<MutableList<Course>> = arrayOf()
  private var mWeek = 0
  private var mDay = 0

  fun setData(data: Array<MutableList<Course>>, week: Int, day: Int) {
    mData = data
    mWeek = week
    mDay = day
    notifyDataSetChanged()
  }

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
    val view = ComposeView(parent.context)
    return object : RecyclerView.ViewHolder(view) {
    }
  }

  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
    val list = mData[position]
    val weekString = " $mWeek "
    val course = list.firstOrNull { it.week2.contains(weekString) } ?: list.lastOrNull()
    val vm = CourseDayItemViewModel(
      course = course,
      multiple = list.size > 1,
      valid = course != null && course.week2.contains(weekString),
      position = position,
      onClick = {
        if (list.isNotEmpty()) {
          fragment.showCourseList(list, mDay, position)
        }
      }
    )
    (holder.itemView as ComposeView).setContent {
      CourseDayItem(vm = vm)
    }
  }

  override fun getItemCount(): Int {
    return mData.size
  }
}
